cirq-core/cirq/ops/uniform_superposition_gate.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2024 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Sequence, TYPE_CHECKING
+
+import numpy as np
+from cirq.ops.common_gates import H, ry
+from cirq.ops.pauli_gates import X
+from cirq.ops import raw_types
+
+
+if TYPE_CHECKING:
+    import cirq
+
+
+class UniformSuperpositionGate(raw_types.Gate):
+    r"""Creates a uniform superposition state on the states $[0, M)$
+    The gate creates the state $\frac{1}{\sqrt{M}}\sum_{j=0}^{M-1}\ket{j}$
+    (where 1<= M <= 2^n), using n qubits, according to the Shukla-Vedula algorithm [SV24].
```

Comment:
```suggestion
    (where $1\leq M \leq 2^n$), using n qubits, according to the Shukla-Vedula algorithm [SV24].
```

Reply:
@NoureldinYosri - Thank you for the suggestion. I made the suggested changes. The format check is passing locally for me now. This should pass now in CI as well.